readthedocs/search/management/commands/reindex_elasticsearch.py

```diff
@@ -58,56 +52,127 @@ def _run_reindex_tasks(self, models, queue):
-                                                 index_name=index_name,
-                                                 new_index_name=new_index_name)
-
-            # Task to run in order to add the objects
-            # that has been inserted into database while indexing_tasks was running
-            # We pass the creation time of latest object, so its possible to index later items
-            missed_index_task = index_missing_objects.si(app_label=app_label,
-                                                         model_name=model_name,
-                                                         document_class=str(doc),
-                                                         index_generation_time=index_time)
-
-            # http://celery.readthedocs.io/en/latest/userguide/canvas.html#chords
-            chord_tasks = chord(header=indexing_tasks, body=post_index_task)
-            if queue:
-                pre_index_task.set(queue=queue)
-                chord_tasks.set(queue=queue)
-                missed_index_task.set(queue=queue)
-            # http://celery.readthedocs.io/en/latest/userguide/canvas.html#chain
-            chain(pre_index_task, chord_tasks, missed_index_task).apply_async(**apply_async_kwargs)
-
-            message = ("Successfully issued tasks for {}.{}, total {} items"
-                       .format(app_label, model_name, queryset.count()))
-            log.info(message)
+        for doc in registry.get_documents(models=[Project]):
+            indexing_tasks = self._get_indexing_tasks(
+                app_label=app_label,
+                model_name=model_name,
+                queryset=queryset,
+                index_name=doc._index._name,
+                document_class=str(doc),
+            )
+            for task in indexing_tasks:
+                task.apply_async(**apply_async_kwargs)
+            log.info(
+                "Tasks issued successfully. model=%s.%s items=%s",
+                app_label, model_name, str(queryset.count())
+            )
+
+    def _reindex_files_from(self, days_ago, queue=None):
+        """Reindex HTML files from versions with recent builds."""
+        since = datetime.now() - timedelta(days=days_ago)
+        queryset = Version.objects.filter(builds__date__gte=since).distinct()
+        app_label = HTMLFile._meta.app_label
+        model_name = HTMLFile.__name__
+        apply_async_kwargs = {
+            'kwargs': {
+                'app_label': app_label,
+                'model_name': model_name,
+            },
+        }
+        if queue:
+            apply_async_kwargs['queue'] = queue
+
+        for doc in registry.get_documents(models=[HTMLFile]):
+            apply_async_kwargs['kwargs']['document_class'] = str(doc)
+            for version in queryset.iterator():
+                files_id = list(
+                    HTMLFile.objects
+                    .filter(version=version)
+                    .values_list('pk', flat=True)
+                )
```

Comment:
I don't think the number of files from a version would be really huge to cause an EOM error, but let me know if you want me to use a limit here.

Reply:
actually I'm going to add a limit here using the same chunk size.